tools/webtransport/h3/webtransport_h3_server.py

```diff
@@ -225,6 +312,7 @@ def __init__(self, session: WebTransportSession,
                  callbacks: Dict[str, Any]) -> None:
         self._session = session
         self._callbacks = callbacks
+        self._allow_calling_session_closed = True
```

Comment:
How about having `_session_closed` flag in WebTransportH3Protocol or WebTransportSession. It is set when close() is called. session_closed() handler is bypassed when the _session_closed flag is set.

(I would like to keep this handler class a thin wrapper of an actual handler. Putting state management in WebTransportH3Protocol or WebTransportSession will help keeping this wrapper simple)

Reply:
`_session_closed` would need to be updated when `session_closed` is called (it's also called in `quic_event_received`). How would you update it?

Reply:
How about the latest change?